UI/window-basic-main-dropfiles.cpp

```diff
@@ -105,10 +110,19 @@ void OBSBasic::AddDropSource(const char *data, DropType image)
 		name = QUrl::fromLocalFile(QString(data)).fileName();
 		type = "browser_source";
 		break;
+	case DropType_Source:
+		json_settings = obs_data_create_from_json_file(data);
+		settings = obs_data_get_obj(json_settings, "settings");
+		name = obs_data_get_string(json_settings, "name");
+		type = obs_data_get_string(json_settings, "id");
+		break;
 	}
 
-	if (!obs_source_get_display_name(type))
+	if (!obs_source_get_display_name(type)) {
+		obs_data_release(settings);
```

Comment:
Please don't add this (`obs_data_release(settings);`) in the same commit; make a separate commit ("UI: Fix drag&drop memory leak" or similar)

Reply:
That would make more sense to do wouldn't it...

Reply:
Ah...sorry, don't know why I thought you meant pr / commit as opposed to just commit. Now I'm proper confused.

Reply:
Ideally both (separate PR and commit)

Reply:
There shouldn't be a merge conflict within the if block if I separate these out right?

Reply:
There'll likely not be a merge conflict if we follow these steps:
1. you make a separate PR with the leak fix
2. I merge that PR
3. you rebase this PR on top of the fixed code

Reply:
Alright, well, got step 1. out of the way, I'll rebase after I actually finish my lunch.

Reply:
Ready for step 3

Reply:
Doing a local build just to triple check*, I'll be pushing in maybe 15~